cpp/src/lance/format/schema.cc

```diff
@@ -477,7 +477,7 @@ ::arrow::Result<std::shared_ptr<Schema>> Schema::Project(
 
     auto actual_field = GetField(components[0]);
     if (!actual_field) {
-      return ::arrow::Status::Invalid("Field {} does not exist.", name);
+      continue;
```

Comment:
Is this safe? Is this case always legitimate?

Reply:
Arrow strips out the partitioned columns, so the schema of these partitioned columns do not exist in the file.
As a result, from the file reader aspect, there is no reliable way to decide whether these columns are partitioned, or some columns that do not exist. 

We can add one test to verify it.